subworkflows/local/experimental/main.nf

```diff
@@ -30,11 +30,26 @@ workflow EXPERIMENTAL {
     // DIFFERENTIAL ANALYSIS BLOCK
     // ----------------------------------------------------
 
+    // parse optional input files that affect the normalization
+    // TODO we should consider to put this kind of stuff in a separate data handling / preprocessing / normalization block
+    if (params.control_features) {
+        ch_control_features = Channel.of([ [ "id": params.study_name  ], file(params.control_features, checkIfExists: true)]).first()
+    } else {
+        ch_control_features = [[],[]]
+    }
+    if (params.transcript_length_matrix) {
+        ch_transcript_lengths = Channel.of([ [ "id": params.study_name  ], file(params.transcript_length_matrix, checkIfExists: true)]).first()
+    } else {
+        ch_transcript_lengths = [[],[]]
+    }
+
```

Comment:
As this is Deseq2 specific, wouldn't it be better to move it into the differential_analysis subworkflow?

Reply:
it is good to call params within the subworkflows? I thought that should be avoided for a better encapsulation of the subworflow?
Also, not now, but in the future, I think it make sense to have two separated blocks with DESEQ2_NORM dealing the normalization of the data (so in the data processing block) that will use these control/transcript length features, and DESEQ2_DIFFERENTIAL running the differential analysis

Reply:
Yes I see your point, but I would argue that we have a better encapsulation, when a subworkflow can be so transparently changed (i.e. by adding a new module) that there is no need for making any changes to the calling workflow.